src/nnet3/nnet-component-itf.cc

```diff
@@ -96,6 +96,8 @@ Component* Component::NewComponentOfType(const std::string &component_type) {
     ans = new ConvolutionComponent();
   } else if (component_type == "MaxpoolingComponent") {
     ans = new MaxpoolingComponent();
+  } else if (component_type == "Maxpooling3dComponent") {
```

Comment:
Still going through the new commit, will suggest changes if it is not optimal for 1d/2d cases.